modules/querying/pages/query-optimizer/index.adoc

```diff
@@ -0,0 +1,34 @@
+= Query Optimizer
+:description: Overview of the query optimizer.
+
+GSQL has a built-in query optimizer that can optimize the execution plan of a query when you install the query.
+Currently, the optimizer can optimize the execution plan in the following ways:
+
+* <<_cost_based_optimization>>
+
+== What is the query optimizer?
+The query optimizer optimizes the compiled binary execution plan of a query.
+
+When you xref:query-operations.adoc#_install_query[install a GSQL query], GSQL compiles the query logic into a binary execution plan.
+If your query contains a path pattern, the pattern contains a source vertex set, one mor more path edge patterns, and one or more step vertex sets.
+Similar to the path pattern in GSQL, its binary execution plan also starts from a set of source vertices, traverses through edges in the specified pattern, and finds the target vertices.
+
+However, the actual execution plan of the query might perform path traversals differently than specified in the GSQL query to product results.
+When you install a query, the query optimizer can optimize the execution plan of a query so that it executes more efficiently.
+
+Depending on the type of optimization used, this could allow a query to execute with less time, less memory, or less storage.
```

Comment:
There's a lot of good info in this section, but it might be too much. I think I would try to condense it. For instance, I think it would be sufficient to say something like
```
There may be many equivalent ways to express the same query in GSQL.
The job of the optimizer is to find the best way, or more accurately, avoid the most inefficient execution plans.
As a result, the user queries for the data they're interested in but does not have to worry about the details of how it's retrieved.
```

Reply:
here again, i would not say that we explicitly optimize for less memory or less storage.